--- Old/com/mojang/minecraft/server/MinecraftServer.java	2021-07-18 13:06:06.668282100 +0200
+++ New/com/mojang/minecraft/server/MinecraftServer.java	2021-07-18 17:16:56.841320000 +0200
@@ -11,8 +11,8 @@
 
 public class MinecraftServer implements Runnable, ServerListener {
    private SocketServer socketServer;
-   private Map clientMap = new HashMap();
-   private List clients = new ArrayList();
+   private Map<SocketConnection, Client> clientMap = new HashMap<>();
+   private List<Client> clients = new ArrayList<>();
 
    public MinecraftServer(byte[] ips, int port) throws IOException {
       this.socketServer = new SocketServer(ips, port, this);
