--- Old/com/mojang/minecraft/comm/SocketConnection.java	2021-07-17 13:57:44.396347600 +0200
+++ New/com/mojang/minecraft/comm/SocketConnection.java	2021-07-17 13:58:43.973453300 +0200
@@ -19,7 +19,7 @@
    private ConnectionListener connectionListener;
    private int bytesRead;
    private int totalBytesWritten;
-   private int maxBlocksPerIteration = 3;
+   private int maxBlocksPerIteration = 3; // Not sure where to place this variable.
    private Socket socket;
    private BufferedInputStream in;
    private BufferedOutputStream out;
