--- Old/com/mojang/minecraft/comm/SocketServer.java	2021-07-18 13:06:06.642004300 +0200
+++ New/com/mojang/minecraft/comm/SocketServer.java	2021-07-18 17:16:56.794456800 +0200
@@ -13,7 +13,7 @@
 public class SocketServer {
    private ServerSocketChannel ssc;
    private ServerListener serverListener;
-   private List connections = new LinkedList();
+   private List<SocketConnection> connections = new LinkedList<>();
    protected static Logger logger = Logger.getLogger("SocketServer");
 
    public SocketServer(byte[] ips, int port, ServerListener serverListener) throws IOException {
