--- NonFixedSrc/com/mojang/minecraft/renderer/Tesselator.java	2021-07-15 16:15:41.773533400 +0200
+++ FixedSrc/com/mojang/minecraft/renderer/Tesselator.java	2021-07-16 10:39:12.664375100 +0200
@@ -5,8 +5,8 @@
 import org.lwjgl.opengl.GL11;
 
 public class Tesselator {
-   private static final int MAX_MEMORY_USE = 4194304;
-   private static final int MAX_FLOATS = 524288;
+   private static final int MAX_MEMORY_USE = 4194304; // Not sure where to place this constant.
+   private static final int MAX_FLOATS = 524288; // Not sure where to place this constant.
    private FloatBuffer buffer = BufferUtils.createFloatBuffer(524288);
    private float[] array = new float[524288];
    private int vertices = 0;
