--- NonFixedSrc/com/mojang/minecraft/renderer/Textures.java	2021-07-15 16:15:41.773533400 +0200
+++ FixedSrc/com/mojang/minecraft/renderer/Textures.java	2021-07-16 10:39:12.665371800 +0200
@@ -11,7 +11,7 @@
 import org.lwjgl.util.glu.GLU;
 
 public class Textures {
-   private HashMap idMap = new HashMap();
+   private HashMap<String, Integer> idMap = new HashMap<>();
 
    public int loadTexture(String resourceName, int mode) {
       try {
