--- NonFixedSrc/com/mojang/minecraft/Entity.java	2021-07-15 16:15:41.767549800 +0200
+++ FixedSrc/com/mojang/minecraft/Entity.java	2021-07-16 10:39:12.651410300 +0200
@@ -77,7 +77,7 @@
       float xaOrg = xa;
       float yaOrg = ya;
       float zaOrg = za;
-      List aABBs = this.level.getCubes(this.bb.expand(xa, ya, za));
+      List<AABB> aABBs = this.level.getCubes(this.bb.expand(xa, ya, za));
 
       int i;
       for(i = 0; i < aABBs.size(); ++i) {
