--- Old/com/mojang/minecraft/MinecraftApplet.java	2021-07-18 13:06:06.665290000 +0200
+++ New/com/mojang/minecraft/MinecraftApplet.java	2021-07-18 13:07:19.282368600 +0200
@@ -5,12 +5,15 @@
 import java.awt.Canvas;
 
 public class MinecraftApplet extends Applet {
+
+   private static final long serialVersionUID = -8153740015188474976L;
    private Canvas canvas;
    private Minecraft minecraft;
    private Thread thread = null;
 
    public void init() {
       this.canvas = new Canvas() {
+    	 private static final long serialVersionUID = -8153787515187474976L;
          public void addNotify() {
             super.addNotify();
             MinecraftApplet.this.startGameThread();
