--- Old/com/mojang/minecraft/level/DirtyChunkSorter.java	2021-07-18 13:06:06.642973700 +0200
+++ New/com/mojang/minecraft/level/DirtyChunkSorter.java	2021-07-18 13:07:19.265415300 +0200
@@ -4,7 +4,7 @@
 import com.mojang.minecraft.renderer.Frustum;
 import java.util.Comparator;
 
-public class DirtyChunkSorter implements Comparator {
+public class DirtyChunkSorter implements Comparator<Chunk> {
    private Player player;
    private Frustum frustum;
    private long now = System.currentTimeMillis();
