--- NonFixedSrc/com/mojang/minecraft/level/DirtyChunkSorter.java	2021-07-15 16:15:41.768546800 +0200
+++ FixedSrc/com/mojang/minecraft/level/DirtyChunkSorter.java	2021-07-16 10:39:12.654402800 +0200
@@ -4,7 +4,7 @@
 import com.mojang.minecraft.renderer.Frustum;
 import java.util.Comparator;
 
-public class DirtyChunkSorter implements Comparator {
+public class DirtyChunkSorter implements Comparator<Chunk> {
    private Player player;
    private Frustum frustum;
    private long now = System.currentTimeMillis();
