--- Old/com/mojang/minecraft/level/tile/GrassTile.java	2021-07-18 13:06:06.663298000 +0200
+++ New/com/mojang/minecraft/level/tile/GrassTile.java	2021-08-07 19:26:32.086198000 +0200
@@ -3,10 +3,13 @@
 import com.mojang.minecraft.level.Level;
 import java.util.Random;
 
+/**
+ * This class represents a tile of grass.
+ */
 public class GrassTile extends Tile {
    protected GrassTile(int id) {
       super(id);
-      this.tex = 3;
+      tex = 3;
    }
 
    protected int getTexture(int face) {
