--- Old/com/mojang/minecraft/level/tile/GrassTile.java	2021-07-18 13:06:06.663298000 +0200
+++ New/com/mojang/minecraft/level/tile/GrassTile.java	2021-07-18 17:16:56.810077100 +0200
@@ -3,6 +3,9 @@
 import com.mojang.minecraft.level.Level;
 import java.util.Random;
 
+/**
+ * This class represents a tile of grass.
+ */
 public class GrassTile extends Tile {
    protected GrassTile(int id) {
       super(id);
