--- Old/com/mojang/minecraft/character/Cube.java	2021-07-18 13:06:06.637989200 +0200
+++ New/com/mojang/minecraft/character/Cube.java	2021-07-18 17:16:56.778835900 +0200
@@ -79,8 +79,8 @@
 
    private void compile() {
       this.list = GL11.glGenLists(1);
-      GL11.glNewList(this.list, 4864);
-      GL11.glBegin(7);
+      GL11.glNewList(this.list, GL11.GL_COMPILE);
+      GL11.glBegin(GL11.GL_QUADS);
 
       for(int i = 0; i < this.polygons.length; ++i) {
          this.polygons[i].render();
