--- Old/com/mojang/minecraft/character/Zombie.java	2021-07-18 13:06:06.639982700 +0200
+++ New/com/mojang/minecraft/character/Zombie.java	2021-07-18 17:16:56.778835900 +0200
@@ -19,7 +19,7 @@
       this.rotA = (float)(Math.random() + 1.0D) * 0.01F;
       this.setPos(x, y, z);
       this.timeOffs = (float)Math.random() * 1239813.0F;
-      this.rot = (float)(Math.random() * 3.141592653589793D * 2.0D);
+      this.rot = (float)(Math.random() * Math.PI * 2.0D);
       this.speed = 1.0F;
    }
 
@@ -56,8 +56,8 @@
    }
 
    public void render(float a) {
-      GL11.glEnable(3553);
-      GL11.glBindTexture(3553, this.textures.loadTexture("/char.png", 9728));
+      GL11.glEnable(GL11.GL_TEXTURE_2D);
+      GL11.glBindTexture(GL11.GL_TEXTURE_2D, this.textures.loadTexture("/char.png", GL11.GL_NEAREST));
       GL11.glPushMatrix();
       double time = (double)System.nanoTime() / 1.0E9D * 10.0D * (double)this.speed + (double)this.timeOffs;
       float size = 0.058333334F;
@@ -70,6 +70,6 @@
       GL11.glRotatef(this.rot * c + 180.0F, 0.0F, 1.0F, 0.0F);
       zombieModel.render((float)time);
       GL11.glPopMatrix();
-      GL11.glDisable(3553);
+      GL11.glDisable(GL11.GL_TEXTURE_2D);
    }
 }
