--- Old/com/mojang/minecraft/level/DirtyChunkSorter.java	2021-07-17 13:57:44.396347600 +0200
+++ New/com/mojang/minecraft/level/DirtyChunkSorter.java	2021-07-17 13:58:43.989022800 +0200
@@ -4,7 +4,7 @@
 import com.mojang.minecraft.renderer.Frustum;
 import java.util.Comparator;
 
-public class DirtyChunkSorter implements Comparator {
+public class DirtyChunkSorter implements Comparator<Chunk> {
    private Player player;
    private Frustum frustum;
    private long now = System.currentTimeMillis();
